opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -549,15 +548,15 @@ def emit(self, record: logging.LogRecord) -> None:
 
         The record is translated to OTel format, and then sent across the pipeline.
         """
-        if not isinstance(self._logger, NoOpLogger):
-            self._logger.emit(self._translate(record))
+        logger = get_logger(record.name, logger_provider=self._logger_provider)
+        if not isinstance(logger, NoOpLogger):
+            logger.emit(self._translate(record))
 
     def flush(self) -> None:
         """
         Flushes the logging output. Skip flushing if logger is NoOp.
         """
-        if not isinstance(self._logger, NoOpLogger):
-            self._logger_provider.force_flush()
+        self._logger_provider.force_flush()
```

Comment:
Why remove the if statement here?

Reply:
There is no longer a single `self._logger` instance to check against. Perhaps we need to keep track of all logger instances (one for each logger name encountered) and do a force_flush() if even a single one is not a NoOpLogger.

Reply:
Got it. It doesn't seem like there would be an issue with flushing a no op either so I think this works fine. Resolved.